lib/drab/browser.ex

```diff
@@ -296,4 +297,161 @@ defmodule Drab.Browser do
     window.history.pushState({}, "", #{Drab.Core.encode_js(url)});
     """)
   end
+
+  @doc """
+  Synchronously set a cookie on the client side
+  
+  Returns tuple {status, return_value}, where status could be :ok, :error or :timeout, and return value contains the output computed by the Javascript or the error message.  
+
+  ### Parameters:
+
+  * `socket`  - the Drab socket
+  * `key`     - the cookie name
+  * `value`   - the cookie value, can be an Elixir data structure
+        
+        
+  ### Options:
+
+  * `max-age` :
+   
+       max-age > 0    - sets the cookie max-age, in seconds
+     
+       max-age == 0   - sets a session cookie
+     
+       max-age < 0    - delete the cookie
+     
+     
+  * `path` - the cookie path, default `"/"`
+  
+  * `encode` - encodes the cookie in Base64. The encoding is needed if you want to store values with comma, semicolon, quotes, etc or Elixir data structures.   
+  
+  
+  ### Example:
+  
+        iex> Drab.Browser.set_cookie(socket, "Items", [%{id: 001, name: "foo"}, %{id: 002, name: "bar"}], max_age: 3*24*60*60, encode: true)
+        {:ok, result}
+  """
+  # @spec exec_js(Phoenix.Socket.t(), String.t(), Keyword.t()) :: result
+  def set_cookie(socket, key, value, options \\ []) do
+    # Options
+    max_age = Keyword.get(options, :max_age, 0)
+    path = Keyword.get(options, :path, "/")
+
+    # Prepare
+    encoded_value = encode_value(value, options)
+    expires = cookie_expires(socket, max_age)
+
+    # Set cookie
+    exec_js(socket, "document.cookie='#{key}=#{encoded_value}; expires=#{expires}; path=#{path};'")
```

Comment:
It is not intuitive, that it deletes **all the other cookies**. It should set the existing one, and leave all other untouched.

Reply:
I'm not sure to understand what do you mean. Why you say that the other cookies are deleted?

Reply:
Forget about it, my mistake, I did not know that `document.cookie = ` is clever enough to set the specific one only (I thought it is setting up the string with all the cookies). It is OK.